project_timesheet_time_control/models/account_analytic_line.py

```diff
@@ -13,13 +15,39 @@ class AccountAnalyticLine(models.Model):
     _inherit = "account.analytic.line"
     _order = "date_time desc"
 
-    date_time = fields.Datetime(default=fields.Datetime.now, copy=False)
+    date_time = fields.Datetime(
+        string="Start Time", default=fields.Datetime.now, copy=False
+    )
+    date_time_end = fields.Datetime(
+        string="End Time",
+        compute="_compute_date_time_end",
+        inverse="_inverse_date_time_end",
+    )
     show_time_control = fields.Selection(
         selection=[("resume", "Resume"), ("stop", "Stop")],
         compute="_compute_show_time_control",
         help="Indicate which time control button to show, if any.",
     )
 
+    @api.depends("date_time", "unit_amount")
+    def _compute_date_time_end(self):
+        hour_uom = self.env.ref("uom.product_uom_hour")
+        for record in self:
+            if record.product_uom_id == hour_uom:
+                record.date_time_end = record.date_time + relativedelta(
+                    hours=record.unit_amount
+                )
+            else:
+                record.date_time_end = False
```

Comment:
I think it should be like this:

```suggestion
                record.date_time_end = record.date_time_end
```

Reply:
included in my latest commit